cni/pkg/install/install.go

```diff
@@ -120,37 +122,29 @@ func (in *Installer) Cleanup() error {
 	installLog.Info("Cleaning up.")
 	if len(in.cniConfigFilepath) > 0 && file.Exists(in.cniConfigFilepath) {
 		if in.cfg.ChainedCNIPlugin {
-			installLog.Infof("Removing Istio CNI config from CNI config file: %s", in.cniConfigFilepath)
-
-			// Read JSON from CNI config file
-			cniConfigMap, err := util.ReadCNIConfigMap(in.cniConfigFilepath)
-			if err != nil {
-				return err
-			}
-			// Find Istio CNI and remove from plugin list
-			plugins, err := util.GetPlugins(cniConfigMap)
-			if err != nil {
-				return fmt.Errorf("%s: %w", in.cniConfigFilepath, err)
-			}
-			for i, rawPlugin := range plugins {
-				plugin, err := util.GetPlugin(rawPlugin)
-				if err != nil {
-					return fmt.Errorf("%s: %w", in.cniConfigFilepath, err)
-				}
-				if plugin["type"] == "istio-cni" {
-					cniConfigMap["plugins"] = append(plugins[:i], plugins[i+1:]...)
-					break
+			// For chained mode, other processes are potentially watching/writing
+			// the shared node config during install/uninstall/whatever.
+			//
+			// This can lead to contentions during uninstall in the worst case,
+			// so if removal fails, retry a few times before giving up.
+			//
+			// tryRemoveChainedConfig will only return nil on a successful write.
+			// nolint: gosec
+			r := rand.Intn(3)
+			for i := 0; i < 3; i++ {
+				if err := in.tryRemoveChainedConfig(); err != nil {
+					installLog.Warnf("Detected contention removing Istio CNI config from shared node CNI file: %s, error was %s, retrying...",
+						in.cniConfigFilepath, err)
+					time.Sleep(time.Duration(r) * time.Second)
```

Comment:
2s is pretty long to sleep, and can return 0s. Seems like we could be a lot more aggressive here